src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -582,5 +655,227 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
 
             return parameters.Count - 1;
         }
+
+        protected (IEnumerable<T> parameters, IEnumerable<SyntaxToken> separators) UpdateDeclarationBase<T>(
```

Comment:
can these be immutable arrays?  i really don't like seeing Lists/IEnumerables because i can't reason aobut if they stay the same or changes.

Reply:
Fixed in the next review.